VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -1352,61 +1354,34 @@ private AccessibilityManager getAccessibilityManager() {
      */
     private boolean getTouchExploreEnabled() {
         boolean ret = false;
-        AccessibilityManager am = getAccessibilityManager();
-        if (am != null && am.isTouchExplorationEnabled()) {
-            ret = true;
-        }
-        return ret;
-    }
 
-    /**
-     * @param am The accessibility manager to unregister from
-     * @return True if successful
-     */
-    private boolean unregisterTouchExploreChangeListener(@NonNull AccessibilityManager am) {
-        boolean ret = false;
-        if (touchExplorationStateChangeListener == null) {
-            ret = true;
-        }
-        else if (am.removeTouchExplorationStateChangeListener(touchExplorationStateChangeListener)) {
-            touchExplorationStateChangeListener = null;
-            ret = true;
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            AccessibilityManager am = getAccessibilityManager();
+            if (am != null && am.isTouchExplorationEnabled()) {
+                ret = true;
+            }
         }
-        return ret;
-    }
 
-    /**
-     * If a listener is previously registered, unregisters then registers this one
-     * @param am The accessibility manager to unregister from/register to
-     * @param listener The new listener
-     * @return True if successful
-     */
-    private boolean registerTouchExploreChangeListener(@NonNull AccessibilityManager am, @NonNull AccessibilityManager.TouchExplorationStateChangeListener listener) {
-        boolean ret = false;
-        if (unregisterTouchExploreChangeListener(am) && am.addTouchExplorationStateChangeListener(listener)) {
-            touchExplorationStateChangeListener = listener;
-            ret = true;
-        }
         return ret;
     }
 
-    protected boolean setTouchExploreChangeListener(@Nullable AccessibilityManager.TouchExplorationStateChangeListener listener) {
-        boolean ret = false;
+    @TargetApi(Build.VERSION_CODES.KITKAT)
+    protected void setTouchExploreChangeListener(@Nullable AccessibilityManager.TouchExplorationStateChangeListener listener) {
         if (listener != touchExplorationStateChangeListener) {
             AccessibilityManager am = getAccessibilityManager();
             if (am != null) {
-                if (listener != null) {
-                    ret = registerTouchExploreChangeListener(am, listener);
+
+                if (touchExplorationStateChangeListener != null) {
+                    am.removeTouchExplorationStateChangeListener((AccessibilityManager.TouchExplorationStateChangeListener) touchExplorationStateChangeListener);
                 }
-                else {
-                    ret = unregisterTouchExploreChangeListener(am);
+
+                if (listener != null) {
```

Comment:
TouchExplorerState change listener is a bit different than other listeners in that it is additive. In other words, adding one doesn't remove the old one.

We need to be careful and only have one in there at a time. So, the code is set up so that if we have a previous one set, we remove that one (unregistering). But if the parameter is non null, we add the new one. This logic only gets hit if there is a difference from the old parameter to the new one.

From a black box perspective, this has the behavior of registering if the parameter is non null but registering if it is null. If the parameter is identical to the last one, it results in a no-op.